samples/CommunityToolkit.Maui.Markup.Sample/HotReloadHandler.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace CommunityToolkit.Maui.Markup.Sample;
+
+class HotReloadHandler : ICommunityToolkitHotReloadHandler
+{
+	public async void OnHotReload(IReadOnlyList<Type> types)
+	{
+		if (Application.Current?.MainPage is not Page mainPage)
+		{
+			return;
+		}
+
+		foreach (var type in types)
+		{
+			if (type.IsSubclassOf(typeof(Page)))
+			{
+				if (mainPage is AppShell shell)
+				{
+					var visiblePage = shell.CurrentPage;
+
+					await mainPage.Dispatcher.DispatchAsync(async () =>
+					{
+						await Shell.Current.GoToAsync(type.Name, false);
+						Shell.Current.Navigation.RemovePage(visiblePage);
+					});
+
+					break;
+				}
+				else
+				{
+					if (TryGetModalStackPage(out var modalPage))
+					{
+						await mainPage.Dispatcher.DispatchAsync(async () =>
+						{
+							await mainPage.Navigation.PopModalAsync(false);
+							await mainPage.Navigation.PushModalAsync(modalPage, false);
```

Comment:
does it clear all the user inputs on the page?

Reply:
This does, yes. 

I've found that saving state is the hardest part of handling C# Hot Reload. 

The .NET MAUI Engineering team really needs to implement C# Hot Reload at the framework-level.

Reply:
The reload basically starts from scratch? Does it still not work if there is bindings?

@drasticactions what/why does C# behave differently and what can be done about it? Should bindings work?

Not sure if general C# reloading will fix things in core maui either. Xaml reloading is very different if I understand things correctly.

Reply:
@brminnick @VincentH-Net MAUI does have some C# Hot Reload framework support built in, added by Clancey for Comet. Hot Reloadable controls, like ContentPage here, implement IHotReloadableView: https://github.com/dotnet/maui/blob/db60162cc3f8010e008f2ff129e3737f5c33c5fb/src/Controls/src/Core/ContentPage/ContentPage.cs#L101.  No one uses that today (except Comet) and it's unclear how well it works currently, but that core support is there - at least that's the intention of that code. You may want to: Test that support, to see how well it works in practice. If it works well enough, use it. If there are fixable issues, you could potentially PR fixes, helping get the MAUI framework C# Hot Reload support up to speed for .NET8.

Reply:
Good to know! I'll check it out 💯